src/game.rs

```diff
@@ -79,19 +81,120 @@ impl Game {
                 grid.insert(mp, square.clone());
             }
         }
-        debug!("Absolute grid: {:?}", grid);
         grid
     }
 
-    pub fn process_move(&self, m: Move) -> MoveValidity {
-        let heister = m.heister_color;
-        let pos = m.position;
+    fn are_adjacent(my_pos: MapPosition, other_pos: MapPosition) -> bool {
```

Comment:
These should both be references, it'd save you from having to clone them when calling.